tools/deployment/k8s/s3/deploy.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+# AWS CLI Container Image Script
+# Ensure AWS CLI environment setup:
+# - Use official AWS CLI container image (amazon/aws-cli) and ensure curl, tar, gzip, and jq is preinstalled.
+# - Configure AWS credentials using environment variables through various options:
+#   - Put credential files under $HOME/.aws/credentials
+#   - Set AWS_CONFIG_FILE for custom configuration file path.
+#   - Set AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY directly.
+#   - Refer to AWS CLI docs for more authentication options.
+
+# Expected Environment Variables:
+# - AWS_ENDPOINT_URL: Example - "http://minio:9091"
+# - LOG_VIEWER_BUCKET: Example - "log-viewer"
+# - TAG_NAME: Example - "latest" (latest stable release), A GitHub release tag, or <empty> (latest prerelease)
```

Comment:
```suggestion
readonly REQUIRED_ENV_VARS=(
    # Example: "http://minio:9091
    "AWS_ENDPOINT_URL"

    # Example: "log-viewer"
    "LOG_VIEWER_BUCKET"
)

readonly LOG_VIEWER_RELEASE_TAG=${LOG_VIEWER_RELEASE_TAG:-"latest"}
```

* We can use this array in the for loop below when checking for existence of the required env vars.
* Don't think we need the comment for TAG_NAME since the suggested name should be clear and the suggested default should cover the empty/"latest" case.